server/protocols/webadmin/src/test/java/org/apache/james/webadmin/servlet/DomainServletTest.java

```diff
@@ -0,0 +1,341 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.servlet;
+
+import static com.jayway.restassured.RestAssured.given;
+import static com.jayway.restassured.config.EncoderConfig.encoderConfig;
+import static com.jayway.restassured.config.RestAssuredConfig.newConfig;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+
+import org.apache.james.domainlist.api.DomainListException;
+import org.apache.james.domainlist.api.DomainListManagementMBean;
+import org.apache.james.webadmin.Constants;
+import org.apache.james.webadmin.WebAdminServer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.Lists;
+import com.jayway.restassured.RestAssured;
+import com.jayway.restassured.http.ContentType;
+
+public class DomainServletTest {
+
+    public static final Optional<Integer> RANDOM_PORT = Optional.empty();
+
+    private WebAdminServer webAdminServer;
+    private DomainListManagementMBean domainListManagementMBean;
+
+    @Before
+    public void setUp() throws Exception {
+        domainListManagementMBean = mock(DomainListManagementMBean.class);
+        webAdminServer = new WebAdminServer(RANDOM_PORT, new DomainServlet(domainListManagementMBean));
+        webAdminServer.configure(null);
+
+        RestAssured.port = webAdminServer.getPort();
+        RestAssured.config = newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8));
+    }
+
+    @After
+    public void tearDown() {
+        webAdminServer.stop();
+    }
+
+    @Test
+    public void postShouldAccessUnderlyingStorage() throws Exception {
+        String domain = "domain";
+
+        given()
+            .accept(ContentType.JSON)
+            .contentType(ContentType.JSON)
+        .when()
+            .post(Constants.DOMAIN + "/" + domain)
+        .then()
+            .statusCode(200);
+
+        verify(domainListManagementMBean).addDomain(domain);
```

Comment:
Why do you say that I'm testing "how" it does its job ?

Here I test that it returns status code 200, and updated  the underlying storage. It's what it does, not how it does it.